sky/provision/aws/instance.py

```diff
@@ -46,24 +46,17 @@
 # https://aws.amazon.com/ec2/pricing/on-demand/#Data_Transfer_within_the_same_AWS_Region
 
 
-@aws.import_package
-def botocore_config():
-    """Get botocore.config module.
-
-    This is for backward compatibility when `aws` module is imported before the
-    `botocore_config()` function is added in #1702.
-    TODO: remove this function after 0.6.0 release, when new provisioner is
-    used by default.
-    """
-    config = aws.botocore.config
-    return config
-
-
 def _default_ec2_resource(region: str) -> Any:
+    if not hasattr(aws, 'version'):
+        # For backward compatibility, reload the module if the aws module was
+        # imported before and stale. Used for, e.g., a live spot controller
+        # running an older version and a new version gets installed by
+        # `sky spot launch`.
```

Comment:
Probably worth adding something from https://github.com/skypilot-org/skypilot/pull/2675#discussion_r1349797107 to an inline comment.

Reply:
Elaborate the comment. PTAL : )